lib/src/modules/task.cc

```diff
@@ -128,11 +149,145 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+static Module::ProcessingError toModuleProcessingError(const fs::filesystem_error& e) {
+    std::string err_msg = e.what();
+    if (alg::contains(err_msg, "No such file or directory")) {
+      return Module::ProcessingError(lth_loc::format("No such file or directory: {1}", e.path1()));
+    } else if (alg::contains(err_msg, "File exists")) {
+      return Module::ProcessingError(lth_loc::format("A file matching the name of the provided sha already exists"));
+    }
+    return Module::ProcessingError(err_msg);
+}
+
+// Take out common permissions.
+static fs::perms TASK_FILE_PERMS = fs::owner_read | fs::owner_write | fs::group_read;
+
+// Creates the <task_cache_dir>/<sha256> directory for a single
+// task, ensuring that its permissions are readable by
+// the PXP agent owner/group (for unix OSes), writable for the PXP agent owner,
+// and executable by both PXP agent owner and group. Returns the path to this directory.
+// Note that the last modified time of the directory is updated, and that this routine
+// will not fail if the directory already exists.
+static fs::path createCacheDir(const fs::path& task_cache_dir, const std::string& sha256) {
+    auto cache_dir = fs::path(task_cache_dir) / sha256;
```

Comment:
I believe the `fs::path()` construction can be omitted here. Shortening the expression to:
```c++
auto cache_dir = task_cache_dir / sha256;
```